### PR TITLE
Add a bearer token AuthProvider 

### PR DESCRIPTION
**Pull request by vavido**
This adds a new `AuthProvier` implementation to allow authentication with a `Bearer` token.
At the moment, only a implementation for HTTP connections is provided, as there doesn't seem to be a standard way for OAuth authentication with websockets.

### CODE REVIEW

**Comment by vavido** [comment]
Hey everyone, as this is my first pull request here I'd be very gratefull for some feedback, especially if i understood the architecture of the clinets and contribution guidelines correctly.

I'm also not sure about when/how/by whom versions should get bumped, I created a prerelease version for our internal use, but i suppose the version needs to be set to the actual 2.1.0 release before this gets merged.

I'll look into the failed checks right away.

**Comment by vavido** [comment]
Also, this PR should probably close #29, but it seems that I don't have the permission to link issues

File: javascript/lerna.json [hunk 1]
```diff
@@ -16,5 +16,5 @@
       ]
     }
   },
-  "version": "2.0.0"
+  "version": "2.1.0-alpha.1"
```
Comment (yufei-cai) [inline javascript/lerna.json:19]: Version 2.0.0 is not released. Consider the master branch the release candidate of 2.0.0. Please leave the version as-is.

File: javascript/lib/api/src/auth/bearer-auth.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
```
Comment (yufei-cai) [inline javascript/lib/api/src/auth/bearer-auth.ts:2]: The year of the license header should be the year when the file was created.

**Review by yufei-cai — CHANGES_REQUESTED** [review]
LGTM other than a few clerical errors noted in commit comments.

The authentication provider injects the bearer token into the Authorization header. No convenience API is included to obtain the token in the manner of [ClientCredentialsJsonWebTokenSupplier](https://github.com/eclipse/ditto-clients/blob/master/java/src/main/java/org/eclipse/ditto/client/messaging/internal/ClientCredentialsJsonWebTokenSupplier.java) of the java client, but good enough as a start.

Edit: This does not fix #29 because the authentication provider does not address not being able to set headers when opening a plain websocket.

File: javascript/lerna.json (hunk 1 above)
Comment (vavido) [reply javascript/lerna.json:19]: Okay, no problem, allthough I'm a bit confused as the current version on NPM is tagged as 2.0.0 without any indication that it's a release-candidate/pre-release: https://www.npmjs.com/package/@eclipse-ditto/ditto-javascript-client-api_1.0

File: javascript/lib/api/src/auth/bearer-auth.ts (hunk 2 above)
Comment (vavido) [reply javascript/lib/api/src/auth/bearer-auth.ts:2]: I tried that but tslint immediately started comlaining about a missing file headers... I'll fix it in both the config and this file, then.

File: javascript/lerna.json (hunk 1 above)
Comment (ffendt) [reply javascript/lerna.json]: You're right, 2.0.0 is already released, so you could put 2.1.0 here.

**Comment by vavido** [comment]
> No convenience API is included to obtain the token in the manner of [ClientCredentialsJsonWebTokenSupplier](https://github.com/eclipse/ditto-clients/blob/master/java/src/main/java/org/eclipse/ditto/client/messaging/internal/ClientCredentialsJsonWebTokenSupplier.java) of the java client, but good enough as a start.

I thought about this too, as we use Keycloak as an auth provider and the token needs to be refreshed every few minutes. However, we need the same token elsewhere in the application to authenticate with other services as well, so an implementation that actively gets a new token would break the rest of our application.

Maybe I can add a `TokenSupplier` interface that can be implemented by the user in whichever way necessary to get the current auth token? (For example in our case this would simply be getting the current token from vuex store)

File: javascript/lerna.json (hunk 1 above)
Comment (vavido) [reply javascript/lerna.json]: Well, now I already rolled back the version number to 2.0.0 😬 
I hope that's okay as well.

**Comment by yufei-cai** [comment]
> Maybe I can add a TokenSupplier interface that can be implemented by the user in whichever way necessary to get the current auth token? (For example in our case this would simply be getting the current token from vuex store)

That is a good idea.

File: javascript/lerna.json (hunk 1 above)
Comment (yufei-cai) [reply javascript/lerna.json]: Yes, that is fine.

**Review by yufei-cai — APPROVED** [review]

**Comment by vavido** [comment]
@yufei-cai Yes, I think it's best to add the Token provider in this PR, changes coming soon

**Comment by yufei-cai** [comment]
LGTM

**Pull request merged**